ledger/db/src/test_utils/mod.rs

```diff
@@ -2,3 +2,442 @@
+/// * `recipient` - The recipient of the new transaction.
+/// * `amount` - Amount to send.
+/// * `tombstone_block` - The tombstone block for the new transaction.
+/// * `rng` - The randomness used by this function
+pub fn create_transaction_with_amount_and_comparer_and_recipients<
+    L: Ledger,
+    R: RngCore + CryptoRng,
+    O: TxOutputsOrdering,
+>(
+    block_version: BlockVersion,
+    ledger: &mut L,
+    tx_out: &TxOut,
+    sender: &AccountKey,
+    recipients: &[&PublicAddress],
+    value: u64,
+    fee: u64,
+    tombstone_block: BlockIndex,
+    rng: &mut R,
+) -> Tx {
+    let (sender_amount, _) = tx_out.view_key_match(sender.view_private_key()).unwrap();
+
+    let mut transaction_builder = TransactionBuilder::new(
+        block_version,
+        Amount::new(fee, sender_amount.token_id),
+        MockFogResolver::default(),
+        EmptyMemoBuilder::default(),
+    )
+    .unwrap();
+
+    // The first transaction in the origin block should contain enough outputs to
+    // use as mixins.
+    let origin_block_contents = ledger.get_block_contents(0).unwrap();
+    let origin_outputs = &origin_block_contents.outputs;
+
+    // Populate a ring with mixins.
+    let mut ring: Vec<TxOut> = origin_outputs.iter().take(RING_SIZE).cloned().collect();
+    if !ring.contains(tx_out) {
+        ring[0] = tx_out.clone();
+    }
+    let real_index = ring.iter().position(|element| element == tx_out).unwrap();
+
+    // Membership proofs for the full ring.
+    let indexes = ring
+        .iter()
+        .map(|tx_out| ledger.get_tx_out_index_by_hash(&tx_out.hash()).unwrap())
+        .collect::<Vec<u64>>();
+    let membership_proofs = ledger.get_tx_out_proof_of_memberships(&indexes).unwrap();
+
+    let spend_private_key = sender.subaddress_spend_private(DEFAULT_SUBADDRESS_INDEX);
+    let tx_out_public_key = RistrettoPublic::try_from(&tx_out.public_key).unwrap();
+    let onetime_private_key = recover_onetime_private_key(
+        &tx_out_public_key,
+        sender.view_private_key(),
+        &spend_private_key,
+    );
+
+    let input_credentials = InputCredentials::new(
+        ring,
+        membership_proofs,
+        real_index,
+        onetime_private_key,
+        *sender.view_private_key(),
+    )
+    .unwrap();
+    transaction_builder.add_input(input_credentials);
+
+    let amount = Amount {
+        value: value / recipients.len() as u64,
+        token_id: sender_amount.token_id,
+    };
+
+    let rest = value % recipients.len() as u64;
+
+    // Output
+    for (idx, recipient) in recipients.iter().enumerate() {
+        if idx == 0 && rest != 0 {
+            let mut dup_amount = amount;
+            dup_amount.value += rest;
+            transaction_builder
+                .add_output(dup_amount, recipient, rng)
+                .unwrap();
+            continue;
+        }
+
+        transaction_builder
+            .add_output(amount, recipient, rng)
+            .unwrap();
+    }
+
```

Comment:
Leaving this comment mainly to seed the idea for future changes.
I just realized this was code migrated from `transaction/core/test-utils/src/lib.rs` and with this PR so large I would omit changing anything more unless it's specific to new code.


Not sure if the below suggestion feels better to others, but 2 things make this feel like a code smell:
* Conditioning on index 0 (or n-1) inside of a loop.  Often times this can be done more streamlined, thus allowing the compiler to optimize better.  This is only a test file so optimization isn't a concern, but still a _pattern_ to consider.
* Not isolating the single difference of the _amount_ being used in the transactions.  i.e. the `transaction_builder()` call could always be outside of the condition.

```suggestion
    let mut rest = value % recipients.len() as u64;

    for recipient in recipients.iter() {
        let mut tx_amount = amount;
        tx_amount.value += rest;
        rest = 0;
        transaction_builder
            .add_output(tx_amount, recipient, rng)
            .unwrap();
    }
```